src/tabs/tabs.tsx

```diff
@@ -189,23 +189,32 @@ function TabList({
     const selectedId = tabContextValue?.tabStore.useState('selectedId')
 
     React.useLayoutEffect(() => {
-        if (!selectedId || !tabListRef.current) {
-            return
-        }
+        function updateSelectedTabStyle() {
+            if (!selectedId || !tabListRef.current) {
+                return
+            }
+
+            const tabs = tabListRef.current.querySelectorAll('[role="tab"]')
 
-        const tabs = tabListRef.current.querySelectorAll('[role="tab"]')
+            const selectedTab = Array.from(tabs).find(
+                (tab) => tab.getAttribute('id') === selectedId,
+            ) as HTMLElement | undefined
+
+            if (selectedTab) {
+                setSelectedTabElement(selectedTab)
+                setSelectedTabStyle({
+                    left: `${selectedTab.offsetLeft}px`,
+                    width: `${selectedTab.offsetWidth}px`,
+                })
+            }
+        }
 
-        const selectedTab = Array.from(tabs).find(
-            (tab) => tab.getAttribute('id') === selectedId,
-        ) as HTMLElement | undefined
+        updateSelectedTabStyle()
 
-        if (selectedTab) {
-            setSelectedTabElement(selectedTab)
+        window.addEventListener('resize', updateSelectedTabStyle)
```

Comment:
Should this be debounced?

Reply:
I'm guessing we don't want that:

- If the user is in the process of resizing, with a debounce function, they will not immediately see the tab size being resized, which will look weird (like your screenshot)
- We have a couple more instances of similar code to this in Todoist, none of which uses a debounce function.

We can add one if it ends up being an issue.

Reply:
Yeah I think this will cause resizing to be (even more) janky because reading `offsetLeft` and `offsetWidth` will cause a [reflow](https://gist.github.com/paulirish/5d52fb081b3570c81e3a), and the resize event gets called very often. Limiting the redraws during resize should help mitigate this and reduce the work that's being done.

Even on the storybook page which is fairly lightweight, we can alread see its impact during a resize: 
![image](https://github.com/user-attachments/assets/ba73dbff-349a-4306-b4ca-91118ca6251f)